### PR TITLE
fix the device supported of the op unique and unique_with_counts.

### PR DESCRIPTION
**Pull request by wawltor**
fix the device supported of unique op, and use the cpu kernel to replace the kernel of gpu in the op of unique and unque_with_counts.
test=develop
test=document_fix

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_unique.py [hunk 1]
```diff
@@ -68,5 +68,47 @@ def init_config(self):
...
+            'Out': np.array(
+                [2], dtype='int64'),
+            'Index': np.array(
+                [0], dtype='int32')
+        }
+
+    def test_check_output(self):
+        if core.is_compiled_with_cuda():
+            place = core.CUDAPlace(0)
+            self.check_output_with_place(place, atol=1e-3)
```
Comment (LielinJiang) [inline python/paddle/fluid/tests/unittests/test_unique.py]: why atol use 1e-3 here

File: python/paddle/fluid/tests/unittests/test_unique.py [hunk 2]
```diff
@@ -68,5 +68,47 @@ def init_config(self):
...
+        np_tuple.sort(key=lambda x: x[1])
+        target_out = np.array([i[0] for i in np_tuple], dtype='int64')
+        target_index = np.array(
+            [list(target_out).index(i) for i in self.inputs['X']],
+            dtype='int64')
+
+        self.outputs = {'Out': target_out, 'Index': target_index}
+
+    def test_check_output(self):
+        if core.is_compiled_with_cuda():
```
Comment (LielinJiang) [inline python/paddle/fluid/tests/unittests/test_unique.py:108]: 直接使用checkout函数就好了，里面会判断是否用cuda编译的

**Review by kuke — APPROVED** [review]
LGTM for API doc

**Review by nepeplwu — APPROVED** [review]

**Review by Boyan-Liu — APPROVED** [review]
LGTM

**Pull request merged**